packages/dev/loaders/src/BVH/bvhLoader.ts

```diff
@@ -0,0 +1,388 @@
+
+        if (isNaN(offset.x) || isNaN(offset.y) || isNaN(offset.z)) throw new Error("OFFSET: Invalid values");
+
+        node.offset = offset;
+
+        // parse CHANNELS definitions
+        if (node.type != "ENDSITE") {
+            // @ts-ignore
+            tokens = lines.shift()?.trim().split(/\s+/);
+
+            if (tokens[0].toUpperCase() != "CHANNELS") throw new Error("Expected CHANNELS definition");
+
+            const numChannels = parseInt(tokens[1]);
+            // Skip CHANNELS and the number of channels
+            node.channels = tokens.splice(2, numChannels);
+            node.children = [];
+        }
+
+        // read children
+        while (lines.length > 0) {
+            const line = lines.shift()?.trim();
+
+            if (line === "}") {
+                // Finish reading the node
+                return node;
+            } else if (line) {
+                node.children.push(BVHParser._ReadNode(lines, line, node, context));
+            }
+        }
+
+        throw new Error("Unexpected end of file: missing closing brace");
+    }
+
+    /*
+         Recursively reads data from a single frame into the bone hierarchy.
+         The bone hierarchy has to be structured in the same order as the BVH file.
+         keyframe data is stored in bone.frames.
+    
+         - data: splitted string array (frame values), values are shift()ed so
+         this should be empty after parsing the whole hierarchy.
+         - frameTime: playback time for this keyframe.
+         - bone: the bone to read frame data from.
+
+         Note: Position data (specifically Z) is flipped to convert coordinate systems.
+    */
+    private static _ReadFrameData(data: string[], frameTime: number, bone: BVHNode) {
+        if (bone.type === "ENDSITE")
+            // end sites have no motion data
+            return;
+
+        // add keyframe
+        const keyframe = new BVHKeyFrame();
+        keyframe.time = frameTime;
+        keyframe.position = new Vector3();
+        keyframe.rotation = new Quaternion();
+
+        bone.frames.push(keyframe);
+
+        let pitch = 0,
+            yaw = 0,
+            roll = 0;
+
+        // parse values for each channel in node
+        for (let i = 0; i < bone.channels.length; ++i) {
+            const value = data.shift();
+            if (!value) {
+                continue;
+            }
+
+            switch (bone.channels[i]) {
+                case this._XPosition:
+                    keyframe.position.x = parseFloat(value.trim());
+                    break;
+                case this._YPosition:
+                    keyframe.position.y = parseFloat(value.trim());
+                    break;
+                case this._ZPosition:
+                    keyframe.position.z = -parseFloat(value.trim()); // Flip Z position
+                    break;
+                case this._XRotation:
+                    pitch = (parseFloat(value.trim()) * Math.PI) / 180;
```

Comment:
would be possible to do this? (less code)
```suggestion
                    pitch = +value * Math.PI) / 180;
```